### PR TITLE
Ruby Sinatra implementation

### PR DESCRIPTION
**Pull request by pushplay**
WIP PR for discussion

### CODE REVIEW

File: ruby/lib/app.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  })
+end
+
+post '/rest/createAccount' do
+  request.body.rewind
+  request_payload = JSON.parse request.body.read
+  shopper_id = request_payload['shopperId']
+
+  Lightrail::Account.create({
+    shopperId: shopper_id,
```
Comment (pushplay) [inline ruby/lib/app.rb]: Is this the right way to specify the shopperId?

Comment (tjukes) [reply ruby/lib/app.rb]: Should work with either string or symbol hash keys, snake_case or camelCase. When you have a choice the Ruby way is snake case & symbols as hash keys.

File: ruby/lib/app.rb [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      shopperId: request_payload['shopperId'],
+      currency: request_payload['currency'],
+      amount: request_payload['amount']
+  }
+
+  # Try to charge the whole thing to lightrail, and we'll use the amount that would actually get
+  # charged when we do the real transaction.
+  lightrail_share = split_tender_params[:amount]
+
+  split_tender_charge = Lightrail::StripeLightrailSplitTenderCharge.create(split_tender_params, lightrail_share)  # TODO this needs to be simulate instead
```
Comment (pushplay) [inline ruby/lib/app.rb]: I need to update this to use your split tender.

File: ruby/lib/app.rb [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  "account created (or already exists) for shopperId #{static_params[:shopperId]}"
+end
+
+post '/rest/creditAccount' do
+  request.body.rewind
+  request_payload = JSON.parse request.body.read
+  shopper_id = request_payload['shopperId']
+  value = request_payload['value']
+
+  # TODO how do I create a transaction?
```
Comment (pushplay) [inline ruby/lib/app.rb]: ⬆️

Comment (tjukes) [reply ruby/lib/app.rb]: Lightrail::Account.charge({ value: -1, currency: 'ABC', shopper_id: 'someone-who-shops', })

File: ruby/lib/app.rb [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      shopperId: request_payload['shopperId'],
+      currency: request_payload['currency'],
+      amount: request_payload['amount']
+  }
+
+  # Try to charge the whole thing to lightrail, and we'll use the amount that would actually get
+  # charged when we do the real transaction.
+  lightrail_share = split_tender_params[:amount]
+
+  split_tender_charge = Lightrail::StripeLightrailSplitTenderCharge.simulate(split_tender_params, lightrail_share)
```
Comment (tjukes) [inline ruby/lib/app.rb]: You could also use `Lightrail::StripeLightrailSplitTenderCharge.simulate_with_automatic_split(split_tender_params)` here.

**Pull request merged**